src/osd/Watch.cc

```diff
@@ -193,10 +193,10 @@ void Notify::maybe_complete_notify()
     // prepare reply
     bufferlist bl;
     encode(notify_replies, bl);
-    list<pair<uint64_t,uint64_t> > missed;
-    for (auto p = watchers.begin(); p != watchers.end(); ++p) {
-      missed.push_back(make_pair((*p)->get_watcher_gid(),
-				 (*p)->get_cookie()));
+    vector<pair<uint64_t,uint64_t>> missed;
```

Comment:
How about reserving the space? It could save some bytes.

```suggeston
    vector<pair<uint64_t,uint64_t>> missed;
    missed.reserve(watchers.size());
```

Reply:
@rzarzynski  updated and repushed